haystack/components/generators/chat/openai.py

```diff
@@ -168,28 +157,32 @@ def run(self, messages: List[ChatMessage], generation_kwargs: Optional[Dict[str,
         # adapt ChatMessage(s) to the format expected by the OpenAI API
         openai_formatted_messages = self._convert_to_openai_format(messages)
 
-        completion = openai.ChatCompletion.create(
+        chat_completion: Union[Stream[ChatCompletionChunk], ChatCompletion] = self.client.chat.completions.create(
             model=self.model_name,
-            messages=openai_formatted_messages,
+            messages=openai_formatted_messages,  # type: ignore # openai expects list of specific message types
             stream=self.streaming_callback is not None,
             **generation_kwargs,
         )
 
-        completions: List[ChatMessage]
-        if self.streaming_callback:
+        completions: List[ChatMessage] = []
+        # if streaming is enabled, the completion is a Stream of ChatCompletionChunk
+        if isinstance(chat_completion, Stream):
```

Comment:
I resorted to isinstance here because mypy gave me hell. See self.client.chat.completions.create signature